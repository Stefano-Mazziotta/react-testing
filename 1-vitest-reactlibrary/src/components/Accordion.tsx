import React, { useState } from "react";

type AccordionProps = {
  title: string;
  children: React.ReactNode;
};

export function Accordion({ title, children }: AccordionProps) {
  const [show, setShow] = useState(false);
  const handleClick = () => {
    setShow(!show);
  };
  const arrow = show ? "▼" : "→";
  return (
    <section style={{ background: "white", color: "black", padding: "5px" }}>
      <header onClick={handleClick} style={{ cursor: "pointer" }}>
        <h2>{`${arrow} ${title}`}</h2>
      </header>
      {show && <article>{children}</article>}
    </section>
  );
}
