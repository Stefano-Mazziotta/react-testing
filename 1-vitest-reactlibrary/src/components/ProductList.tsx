import { product } from "../models/product";

type productListProps = { items: product[] };

export function ProductList({ items }: productListProps) {
  return (
    <ul>
      {items.map((item) => {
        const { name, price, id } = item;
        return (
          <li key={id} value={name}>
            {name} ${price}
          </li>
        );
      })}
    </ul>
  );
}
