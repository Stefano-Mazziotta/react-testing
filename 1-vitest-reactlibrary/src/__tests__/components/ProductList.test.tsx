import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { ProductList } from "../../components/ProductList";
import { mockProducts } from "../mocks/products.mock";

describe("<ProductList />", () => {
  it("Should render a list of products", () => {
    const { length } = mockProducts;

    render(<ProductList items={mockProducts} />);
    const products = screen.getAllByRole("listitem");

    expect(products).toHaveLength(length);
  });

  it("should match snapshot", () => {
    render(<ProductList items={mockProducts} />);
    expect(screen.getByRole("list")).toMatchSnapshot();
  });
});
