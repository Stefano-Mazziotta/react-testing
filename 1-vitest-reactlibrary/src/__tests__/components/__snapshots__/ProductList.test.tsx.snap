// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ProductList /> > should match snapshot 1`] = `
<ul>
  <li
    value="Jamón Ibérico"
  >
    Jamón Ibérico
     $
    200
  </li>
  <li
    value="Jamón Serrano"
  >
    Jamón Serrano
     $
    100
  </li>
  <li
    value="Jamón Cocido"
  >
    Jamón Cocido
     $
    50
  </li>
</ul>
`;
